### PR TITLE
Update CODEOWNERS and pin down trivy version

### PR DESCRIPTION
**Pull request by thomasyu888**

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1,2 +1,2 @@
 # Global owners of whole repo
-* @andrewelamb @GiaJordan @linglp @BryanFauble @jaymedina @BWMac @thomasyu888
+* @schematic-developers
```
Comment (thomasyu888) [inline .github/CODEOWNERS:2]:
```suggestion
* @Sage-Bionetworks/schematic-developers
```

File: .github/workflows/scan_repo.yml [hunk 2]
```diff
@@ -17,7 +17,7 @@ jobs:
         uses: actions/checkout@v4
 
       - name: Run Trivy vulnerability scanner in repo mode
-        uses: aquasecurity/trivy-action@master
+        uses: aquasecurity/trivy-action@v0.28.0
```
Comment (thomasyu888) [inline .github/workflows/scan_repo.yml:20]:
```suggestion
        uses: aquasecurity/trivy-action@0.28.0
```

**Review by jaymedina — APPROVED** [review]

**Comment by jaymedina** [comment]
It still requests us all individually for review instead of a "schematic_developers" group. That's the expected behavior, right?

**Review by BWMac — APPROVED** [review]
@jaymedina I think it will request the group instead after it is merged 🤞🏻

**Comment by jaymedina** [comment]
[This commit](https://github.com/Sage-Bionetworks/schematic/pull/1531/commits/5703b59cbda06c0c00c0c5b4d53ab162c5e91fef) was made before the review request was made, which I assumed would have a group be the reviewer instead of the individuals. @BWMac

**Pull request merged**